python/sdk/docs/tutorial.rst

```diff
@@ -0,0 +1,666 @@
+          "--upper-bound",
+          required=True,
+          type=float,
+          help="Initial upper bound for picking a random float to send to the coordinator",
+      )
+      args = parser.parse_args()
+
+
+  if __name__ == "__main__":
+      main()
+
+``main()`` is the function that is called by the ``run-participant``
+command and we already have a ``Participant`` class defined.
+
+
+The ``ParticipantABC`` class
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As explained earlier in the
+:ref:`participant lifecycle paragraph <lifecycle>`,
+a participant needs to communicate with the coordinator. ``xain-sdk``
+already implements this logic so all we need to do is implement the
+``xain-sdk.ParticipantABC`` abstract class, which looks like this:
+
+.. code-block:: Python
+
+    from abc import ABC, abstractmethod
+    from typing import TypeVar
+
+    TrainingResult = TypeVar("TrainingResult")
+    TrainingInput = TypeVar("TrainingInput")
+
+    class ParticipantABC(ABC):
+        @abstractmethod
+        def train_round(self, training_input: TrainingInput) -> TrainingResult:
+            raise NotImplementedError()
+
+        @abstractmethod
+        def serialize_training_result(self, training_result: TrainingResult) -> bytes:
+            raise NotImplementedError()
+
+        @abstractmethod
+        def deserialize_training_input(self, data: bytes) -> TrainingInput:
+            raise NotImplementedError()
+
+There are three methods to implement.
+
+The most important one is ``train_round``, which takes any type of
+object (named ``TrainingInput`` for clarity), and returns a result,
+which can also be any type of object (named ``TrainingResult`` for
+clarity as well). **This is the method that the SDK will call to perform
+the training**. The ``training_input`` argument will be the global model
+retrieved from the coordinator. The training result returned by
+``train_round`` will be sent to the coordinator.
+
+Then come the methods used for data (de)serialization:
+
+- ``deserialize_training_input`` is called right after the SDK has
+  downloaded the latest global model from the coordinator. It is used
+  to deserialize the data that will be passed to ``train_round``.
+- ``serialize_training_result`` is its counterpart: it is called by
+  the SDK to serialize the value returned by ``train_round``, so that
+  it can be sent to the coordinator.
+
+.. note::
+
+   The reason these two methods exist is because there is no
+   limitation on the formats that can be used for the communications
+   between the coordinator and the participants. This is how the XAIN
+   FL Platform can handle such a wide variety of Federated Learning
+   use cases: users have full control on what data is exchanged, and
+   how the coordinator performs the aggregation of all this data,
+   although this is out of scope of this document.
+
+Before implementing these methods, let's make our ``Participant``
+inherit from ``xain-sdk.ParticipantABC``:
+
+.. code-block:: Python
```

Comment:
Maybe it makes sense to add the name of the file here. It was not clear to me where to put this code.